### ISSUE TITLE
Different results

### ISSUE THREAD

**Issue by jodevak** [etano/productner#2]
Thank you for your implementation, but i ran the classifier and evaluated using **whole dataset** and got the following results.

                           precision    recall  f1-score   support

clothing, shoes & jewelry       0.17      0.00      0.00     24536
        sports & outdoors       0.00      0.00      0.00     65860
             toys & games       0.08      0.04      0.06     80918
              movies & tv       0.00      0.00      0.00     11303
                     baby       0.00      0.00      0.00      8675
 tools & home improvement       0.07      0.00      0.00     73641
               automotive       0.20      0.00      0.00    108660
           home & kitchen       0.08      0.11      0.09     61362
    arts, crafts & sewing       0.00      0.00      0.00     18658
          office products       0.00      0.00      0.00     28153
                    books       0.00      0.04      0.00        91
 office & school supplies       0.00      0.00      0.00       420
              electronics       0.05      0.19      0.08     48724
                computers       0.00      0.00      0.00       117
cell phones & accessories       0.01      0.01      0.01      7013
             pet supplies       0.05      0.00      0.00     22393
   health & personal care       0.09      0.00      0.00     61348
              cds & vinyl       0.00      0.00      0.00      6891
      musical instruments       0.02      0.00      0.00     19710
                 software       0.00      0.22      0.00       157
  industrial & scientific       0.00      0.00      0.00      9049
               all beauty       0.00      0.03      0.00       950
              video games       0.00      0.00      0.00       241
                   beauty       0.05      0.00      0.01     52040
     patio, lawn & garden       0.02      0.01      0.02     22239
   grocery & gourmet food       0.00      0.00      0.00     25763
          all electronics       0.00      0.00      0.00       338
            baby products       0.00      0.13      0.01      1631
         kitchen & dining       0.00      0.00      0.00       372
          car electronics       0.00      0.00      0.00        39
            digital music       0.00      0.00      0.00         1
         home improvement       0.00      0.00      0.00       489
           amazon fashion       0.00      0.00      0.00       441
               appliances       0.00      0.00      0.00      1488
           camera & photo       0.00      0.00      0.00        74
         purchase circles       0.00      0.00      0.00         6
         gps & navigation       0.00      0.00      0.00        42
mp3 players & accessories       0.00      0.00      0.00        40
  collectibles & fine art       0.00      0.03      0.00        79
            luxury beauty       0.00      0.00      0.00       415
         furniture & dcor       0.00      0.00      0.00        26
                                0.00      0.00      0.00        30

              avg / total       0.07      0.03      0.02    764423

in other words, the f1-score is way more less than .85 as you mentioned. Is there something im missing ?

**Comment by etano** [comment]
Hah, yeah that looks pretty bad.

Probably something has broken in the meantime since I last tried this. Unfortunately, it will probably be awhile before I can get around to fixing it. I can let you know when I do, but if you want it sooner, I'd advise you to try yourself to see where things are going wrong.

What I'd recommend is first limiting it to only a few categories and seeing if that does better. If not, then likely something has changed in the one of the underlying API's (e.g. Keras).

**Comment by jodevak** [comment]
I will start with decreasing number of cats as you mentioned and batch size,epochs.

**Comment by jodevak** [comment]
precision    recall  f1-score   support

clothing, shoes & jewelry   0.000000  0.000000  0.000000      4984
        sports & outdoors   0.000000  0.000000  0.000000     13274
             toys & games   0.303142  1.000000  0.465248     16062
              movies & tv   0.000000  0.000000  0.000000      2253
                     baby   0.000000  0.000000  0.000000      1817
 tools & home improvement   1.000000  0.000069  0.000137     14596

              avg / total   0.367363  0.303156  0.141072     52986

Results for 6 categories :S

**Comment by etano** [comment]
Yeah, not sure what to say other than that it looks like something is broken then. At least with fewer categories you can iterate faster. But for me to fix it, I'd have to also setup from scratch and then start digging into the processed data.

Maybe make sure that the data looks sane when its done preprocessing. If it does, then likely it's an API change in Keras, Tensorflow, or sklearn.

**Comment by jodevak** [comment]
This is the first row in the preprocessed data.

`big dreams purple sequin tiny dancer tutu ballet dance fairy princess costume accessory,big dreams,big dreams purple sequin tiny dancer tutu ballet dance fairy princess costume accessory ,"clothing, shoes & jewelry / girls / clothing, shoes & jewelry / novelty, costumes & more / costumes & accessories / more accessories / kids & baby",B-B I-B O O O O O O O O O O O `

It might be because of the data quality, not sure, just sayin.

And for the API version, do you have a dockarized version, or at least the version number of each API u have used ?

**Comment by etano** [comment]
I do not unfortunately. It would be the most recent version of each from 6 months ago. Maybe this weekend I can try and get it running.

**Comment by jodevak** [comment]
Cool, looking forward to hearing from you :+1:

**Comment by analystMonkey** [comment]
Hi...Were you able to get it to work ?

**Comment by ekinburak** [comment]
@etano Thanks for sharing! Good job! 

Any update on this issue ?

**Comment by etano** [comment]
I just pushed a "fix". It seems the dropout fraction was a bit too high, probably from a time I was experimenting with it. Now it defaults to no dropout.
